src/main/java/de/zalando/aruha/nakadi/domain/BatchItem.java

```diff
@@ -13,6 +12,11 @@
     public BatchItem(final JSONObject event) {
         this.response = new BatchItemResponse();
         this.event = event;
+
+        Optional.
+                of(event.optJSONObject("metadata"))
+                .map(e -> e.optString("eid"))
```

Comment:
Then I think we need to improve the api definition. It was not clear for me if we should reply with empty string of null. For example, details is empty but never null.